### ISSUE TITLE
Multiple generators not working, deleting previous gens' output

### ISSUE THREAD

**Issue by verdverm** [hofstadter-io/hof#23]
When multiple generators are in the same Cue file and are invoked,
the current generator will remove previous generators output because of how shadow directory is being used to clean per generator.

Using the remaining shadow files for cleanup needs to be moved, or we should support the nested, dependent, collections of generators. Or both? Need to think about the process flow and semantics here.

---

#### Workaround

The current work around involves making all but one generator a definition, and then the one that is a value is simply concatenates the "Out" field from the other generators.

```
package site

import (
  "list"

  hof "github.com/hofstadter-io/hof/schema"
  "github.com/hofstadter-io/hofmod-hugo/gen"

  "github.com/hofstadter-io/site/design"
)


HofGenComponents :: gen.ComponentGen & {
  Outdir: "output"
  Components: design.Components
}

HofGenSections :: gen.SectionGen & {
  Outdir: "output"
  Sections: design.Sections
}

HofGenLayouts :: gen.LayoutGen & {
  Outdir: "output"
  Sections: design.Layouts
}

HofGenAll: hof.HofGenerator & {
  Outdir: "output"
  PackageName: ""
  Out: list.FlattenN([
    HofGenComponents.Out,
    HofGenSections.Out,
    HofGenLayouts.Out,
  ], 1)
}
```

**Comment by verdverm** [comment]
Also, let's add nested generators that exist in the schema already, with this work

**Comment by verdverm** [comment]
Fixed in v0.3.4, just the bug, not the nested generators
